src/features/collections/CollectionDetail.tsx

```diff
@@ -1015,15 +1015,27 @@ const BooleanFilterControls = ({
     setSelectValue(event.target.value as string);
   };
 
+  // In order not to create a dependency loop when filter changes within the below effect,
+  // use a filter ref
+  const filterRef = useRef(filter);
+  filterRef.current = filter;
+
   useEffect(() => {
     let value;
     if (selectValue === 'true') {
       value = 1;
     } else if (selectValue === 'false') {
       value = 0;
     }
-    dispatch(setFilter([collectionId, context, column, { ...filter, value }]));
-  }, [selectValue, collectionId, context, column, filter, dispatch]);
+    dispatch(
+      setFilter([
+        collectionId,
+        context,
+        column,
+        { ...filterRef.current, value },
+      ])
+    );
+  }, [selectValue, collectionId, context, column, dispatch]);
```

Comment:
fixes infinite render loop

Reply:
Oh very nice thank you for catching that